From: Ismael Luceno <ismael@sourcemage.org>

Fix MAJOR and MINOR macros on non-glibc linux systems.

All known linux libc implementations provide `major' and `minor' macros in the
`<sys/sysmacros.h>' header.

Upstream-Status: Pending

--- a/src/lockdev.c
+++ b/src/lockdev.c
@@ -123,13 +123,10 @@
 #include <sys/param.h>
 #include <sys/stat.h>
 #include <sys/file.h>
-#if defined (__GNU_LIBRARY__)
-# include <sys/sysmacros.h>
-# define MAJOR(dev) gnu_dev_major (dev)
-# define MINOR(dev) gnu_dev_minor (dev)
-#else
-#  error "put here a define for MAJOR and MINOR"
-#endif
+
+#include <sys/sysmacros.h>
+#define MAJOR(dev) major(dev)
+#define MINOR(dev) minor(dev)
 
 #include "lockdev.h"
 
